File: db-python-ex/example/DB_01_ConnectMySQL.py
import mysql.connector

# MySQL 데이터베이스에 접속
connection = mysql.connector.connect(
    host='localhost',
    port='3307',
    user='root',
    password='mysql',
    database='world'
)

# 커넥션 상태 확인
if connection.is_connected():
    print("MySQL DB 연결 성공")
else:
    print("MySQL DB 연결 실패")
    exit(100)

# 접속된 데이터베이스에서 쿼리 실행
cursor = connection.cursor()
cursor.execute("SELECT * FROM city LIMIT 100")
result = cursor.fetchall()

# 쿼리 결과 출력
for row in result:
    print(row)

# 연결 종료
cursor.close()
connection.close()


connection.close()
exit(0)

File: db-perl-ex/src/DB_01_ConnectMySQL.pl
use DBI;

# MySQL 데이터베이스 접속 정보
my $host = "localhost";
my $port = "3307";
my $user = "root";
my $password = "mysql";
my $database = "world";

# MySQL 데이터베이스에 접속
# MySQL 8.x부터는 아래의 설정을 해야함.
# alter user 'root' identified with 'mysql_native_password' by 'mysql';
my $dsn = "DBI:mysql:database=$database;host=$host;port=$port";
my $dbh = DBI->connect($dsn, $user, $password) or die "Cannot connect to the database: $DBI::errstr";

# SELECT 쿼리 실행
my $sql = "SELECT * FROM city LIMIT 10";
my $sth = $dbh->prepare($sql);
$sth->execute();

# 결과 출력
print "result 1 assign column\n";
while (my $row = $sth->fetchrow_hashref) {
    print "Column1: " . $row->{ID} . ", Column2: " . $row->{Name} . "\n";
}

$sth->execute();
print "result 2 array\n";
while (my @row = $sth->fetchrow_array) {
    print join(",", @row) . "\n";
}

# 연결 종료
$sth->finish;
$dbh->disconnect;